File: app/src/main/java/ru/mammoth70/wherearetheynow/AboutBox.java
package ru.mammoth70.wherearetheynow;

import androidx.appcompat.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

public class AboutBox extends DialogFragment {
    // Класс создаёт диалоговое окно About.
    public final static String DIALOG_MESSAGE= "DIALOG_MESSAGE";
    public final static String DIALOG_TITLE= "DIALOG_TITLE";

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(requireActivity(),R.style.AboutDialogStyle);
        builder.setIcon(R.mipmap.ic_launcher_round);
        builder.setPositiveButton(R.string.ok, (dialog, id) -> {
            // OK
        });
        if (getArguments() != null) {
            builder.setView(R.layout.dialog_about);
            builder.setTitle(getArguments().getString(DIALOG_TITLE));
            builder.setMessage(getArguments().getString(DIALOG_MESSAGE));
        }
        return builder.create();
    }

}

File: app/src/main/java/ru/mammoth70/wherearetheynow/AboutBox.kt
package ru.mammoth70.wherearetheynow

import android.app.Dialog
import android.os.Bundle
import android.text.method.LinkMovementMethod
import android.widget.TextView
import androidx.fragment.app.DialogFragment
import com.google.android.material.dialog.MaterialAlertDialogBuilder

class AboutBox : DialogFragment() {
    // Диалоговое окно About.

    companion object {
        const val ABOUT_MESSAGE = "ABOUT_MESSAGE"
        const val ABOUT_TITLE = "ABOUT_TITLE"
    }


    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        // Функция создаёт диалоговое окно About.

        val customView = layoutInflater.inflate(R.layout.dialog_about, null)
        val textView = customView.findViewById<TextView>(R.id.about_text_content)

        textView.text = arguments?.getString(ABOUT_MESSAGE)
        textView.movementMethod = LinkMovementMethod.getInstance()

        return MaterialAlertDialogBuilder(requireActivity())
            .setIcon(R.mipmap.ic_launcher_round)
            .setTitle(arguments?.getString(ABOUT_TITLE))
            .setView(customView)
            .setPositiveButton(R.string.ok, null)
            .create()
    }

}